Diffusion/Common/Views/PromptTextField.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  PromptTextField.swift
+//  Diffusion-macOS
+//
+//  Created by Dolmere on 22/06/2023.
+//  See LICENSE at https://github.com/huggingface/swift-coreml-diffusers/LICENSE
+//
+
+import SwiftUI
+import Combine
+import StableDiffusion
+
+struct PromptTextField: View {
+    @State private var output: String = ""
+    @State private var input: String = ""
+    @State private var typing = false
+    @State private var tokenCount: Int = 0
+    @State var isPositivePrompt: Bool = true
+    @State private var tokenizer: BPETokenizer?
+    @State private var currentModelVersion: String = ""
+
+    @Binding var textBinding: String
+    @Binding var model: String // the model version as it's stored in Settings
+
+    private let maxTokenCount = 77
+
+    private var modelInfo: ModelInfo? {
+        ModelInfo.from(modelVersion: $model.wrappedValue)
+    }
+    
+    private var filename: String? {
+        let variant = modelInfo?.bestAttention ?? .original
+        return modelInfo?.modelURL(for: variant).lastPathComponent
+    }
+    
+    private var downloadedURL: URL? {
+        if let filename = filename {
+            return PipelineLoader.models.appendingPathComponent(filename)
+        }
+        return nil
+    }
+    
+    private var packagesFilename: String? {
+        (filename as NSString?)?.deletingPathExtension
+    }
+    
+    private var compiledURL: URL? {
+        if let packagesFilename = packagesFilename {
+            return downloadedURL?.deletingLastPathComponent().appendingPathComponent(packagesFilename)
+        }
+        return nil
+    }
+    
+    private var textColor: Color {
+        switch tokenCount {
+        case 0...65:
+            return .green
+        case 66...75:
+            return .orange
+        default:
+            return .red
+        }
+    }
+    
+    // macOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: Binding<String>) {
+         _textBinding = text
+         self.isPositivePrompt = isPositivePrompt
+        _model = model
+    }
+    
+    //iOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: String) {
+        _textBinding = text
+        self.isPositivePrompt = isPositivePrompt
+        _model = .constant(model)
+    }
+
+    var body: some View {
+        VStack {
+            #if os(macOS)
+            TextField(isPositivePrompt ? "Positive prompt" : "Negative Prompt", text: $textBinding,
+                      axis: .vertical)
+                .lineLimit(20)
+                .textFieldStyle(.squareBorder)
+                .listRowInsets(EdgeInsets(top: 0, leading: -20, bottom: 0, trailing: 20))
+                .foregroundColor(textColor == .green ? .primary : textColor)
+                .frame(minHeight: 30)
+            if (modelInfo != nil && tokenizer != nil) {
```

Comment:
```suggestion
            if modelInfo != nil && tokenizer != nil {
```

nit: I usually prefer `if` in Swift to go without the parenthesis unless they are necessary or helpful. Feel free to disregard if you think it's clearer with them.

Reply:
We should probably create a contributor's style guide document with these little preferences. It'd be optional to comply of course. Overkill?